### ISSUE TITLE
Support for returning HTTP status code from EventGrid-triggered function

### ISSUE THREAD

**Issue by kzu** [Azure/azure-functions-eventgrid-extension#85]
According to the [docs on delivery and retries](https://docs.microsoft.com/en-us/azure/event-grid/delivery-and-retry#failure-codes) 
for EventGrid, the returned HTTP status code can have a significant (and useful!) effect on the retry policy. It would be quite useful 
for a function using the EventGrid binding to be able to report such status codes, yet the documentation only showcases `void` 
returning implementations.

How would a function, for example, report a `503 Service Unavailable` for a downstream processor so that the `Retry after 30 seconds 
or more` applies automatically?

- Function App version: 2.0
- C# web app. 

NOTE: trying to return `Task<IActionResult>` or `Task<HttpResponseMessage>` results in an error like:

```
The '[NAME]' function is in error: Microsoft.Azure.WebJobs.Host: Error indexing method '[NAME]'. Microsoft.Azure.WebJobs.Host: Cannot bind parameter '$return' to type [TYPE]&. Make sure the parameter Type is supported by the binding. If you're using binding extensions (e.g. Azure Storage, ServiceBus, Timers, etc.) make sure you've called the registration method for the extension(s) in your startup code (e.g. builder.AddAzureStorage(), builder.AddServiceBus(), builder.AddTimers(), etc.).
```
(where `[NAME]` is the name of the function and `[TYPE]` is either `HttpResponseMessage` or `IActionResult`). 

> NOTE: for `IActionResult`, I did add a call to `Configure` to add MVC:

```
        public void Configure(IWebJobsBuilder builder)
        {
            ...
            builder.Services.AddMvcCore();
        }
```

#### Repro steps

Provide the steps required to reproduce the problem:

1. Create an Azure Functions app in VS
2. Select an EventGrid trigger function and replace the code with:

```
        [FunctionName("return-status-code")]
        public static Task<HttpResponseMessage> RunAsync([EventGridTrigger] EventGridEvent e)
            => Task.FromResult(new HttpResponseMessage(HttpStatusCode.ServiceUnavailable));
```

3. F5 and see the error:

```
[10/6/2020 18:46:02] Microsoft.Azure.WebJobs.Host: Error indexing method 'return-status-code'. Microsoft.Azure.WebJobs.Host: Cannot bind parameter '$return' to type HttpResponseMessage&. Make sure the parameter Type is supported by the binding. If you're using binding extensions (e.g. Azure Storage, ServiceBus, Timers, etc.) make sure you've called the registration method for the extension(s) in your startup code (e.g. builder.AddAzureStorage(), builder.AddServiceBus(), builder.AddTimers(), etc.).
[10/6/2020 18:46:02] Error indexing method 'return-status-code'
```

#### Expected behavior

Things just work, EventGrid gets the status code from the response and uses that to apply the retry policy as documented.

#### Known workarounds

Use a webhook subscription/function instead of an EventTrigger function :(

**Comment by kirmak** [comment]
It would be good to have a way to control dead-lettering and retry behavior from EventGrid-triggered function. Thanks!

**Comment by kzu** [comment]
Yes, of course it is a feature request ;). Also, yes, since you're *strongly* recommending the use of the Event Grid Trigger, it should offer feature partity with the HTTP one. I mostly care for the mentioned dead-lettering and retry behavior.

**Comment by Kapkan-Lesion** [comment]
@fabiocav Following up on what Kzu mentioned, are there any plans for Azure to offer users the ability to customize EventGrid Trigger's responses (instead of using HTTP trigger for downstream services)? Appreciate your help in advance.

**Comment by greigs** [comment]
I'm also surprised that there is no way to respond to event grid with a failure response, so that the dead lettering can kick in.  Please implement this feature!

**Comment by fabiocav** [comment]
@pragnagopa this should be reviewed so we can define the priority of the work.

**Comment by MikeChristensen** [comment]
I'm curious what the current behavior is.  If the trigger throws _any_ sort of exception, is the message tried again later?  Does it give up after x attempts?  Is there a way to go back and reprocess deadlettered items?  Agreed, this feature really needs to be implemented!

**Comment by Sreini** [comment]
This is of interest to me. From what I understand I cannot have an EventGrid output binding and an HTTP response at the same time?

**Comment by mtvenkatramani** [comment]
Is there any update on implementing this feature - ability to return status codes to event grid?
